crates/ruff/tests/lint.rs

```diff
@@ -5801,3 +5801,65 @@ fn future_annotations_preview_warning() {
     ",
     );
 }
+
+/// Test that our end-of-file handling is the same for all output formats.
+///
+/// There are only four lines in the input (with a trailing newline), but we used to report the
+/// `unexpected EOF` syntax error as being at line 5, column 1. We should now report line 4, column
+/// 2 in all output formats.
```

Comment:
Okay, now I'm a bit confused. 

I assumed that the issue here was that we showed 5:1 for a file with no line break at the end of the file:


```
(
'''unterminated string
'x'
)
```

Here, I expect Ruff to use 4:2 as the location. 

But I'm not so sure if this is the right fix if the file does end with a new line. I tested both textmate and VS code and both allow me to navigate to `5:1` for

```
(
'''unterminated string
'x'
)

```

which is why I think it's more accurate if we do report 5:1 if the offset points to the end of file and we should instead fix the rendering in annotated snippet.